### DISCUSSION TITLE
How to solve disallowed_useragent on FirebaseAuthentication.signInWithGoogle() ?

### DISCUSSION THREAD

**Discussion by phoheisel** [capawesome-team/capacitor-firebase#312] [Q&A] [answered]
I got some user feedback that they can't login through google sign in.
From a screenshot, it seems they get redirected to a web view with a 403: disallowed_useragent

When I do test on my device, sign in with google opens a native kind of overlay which doesn't look like a web view. My guess, this is up to the android version?

The suggested solution from google says don't use a web view:
https://developers.google.com/identity/protocols/oauth2/web-server#authorization-errors-disallowed-useragent

How can I influence the FirebaseAuthentication.signInWithGoogle() to not use a web view? I also did find some suggestions to just use another user agent. Is that a recommended fix?

User Agent fix: https://stackoverflow.com/questions/73993939/getting-error-403-disallowed-useragent-on-ionic-capacitor-inappbrowser-google-l

Thanks for any thoughts!

**Comment by robingenz** [comment, marked as answer]
> How can I influence the FirebaseAuthentication.signInWithGoogle() to not use a web view?

This is handled by the Google Play Services Auth SDK. I assume it falls back to Webview if the device is too old. You can't change this behaviour as far as I know.

> Is that a recommended fix?

To be honest, I have never had this problem and therefore do not know this fix. Feel free to try it out and update this discussion.
